### PR TITLE
Devoncarew add lib api

### LINKED ISSUE #6: perhaps consumable as a library as well?
Perhaps make this consumable as a library as well? So, most of the code in lib/, with a small executable (parsing args) in bin/.

### PR DESCRIPTION
**Pull request by devoncarew**
Fix #6, move much of the `bin/` code into `lib/` and reference that from `bin/`. @sethladd, let me know what you think.

### CODE REVIEW

File: lib/simple_http_server.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (path == null) path = Directory.current.path;
+
+    var handler = createStaticHandler(Directory.current.path,
+        defaultDocument: 'index.html');
+
+    var pipeline = const Pipeline()
+        .addMiddleware(logRequests())
+        .addHandler(handler);
+
+    return io.serve(pipeline, 'localhost', port).then((HttpServer server) {
```
Comment (sethladd) [inline lib/simple_http_server.dart]: feel free to use async and await :)

**Comment by sethladd** [comment]
I can't merge this. But I will once I can. :)

**Comment by devoncarew** [comment]
It's merged to master and changed over to async await -

**Comment by sethladd** [comment]
:rocket:

**Pull request merged**